packages/playwright-core/src/server/registry/index.ts

```diff
@@ -1267,34 +1290,51 @@ export class Registry {
           const descriptor = descriptors.find(d => d.name === browserName);
           if (!descriptor)
             continue;
-          const usedBrowserPath = descriptor.dir;
-          const browserRevision = parseInt(descriptor.revision, 10);
-          // Old browser installations don't have marker file.
-          // We switched chromium from 999999 to 1000, 300000 is the new Y2K.
-          const shouldHaveMarkerFile = (browserName === 'chromium' && (browserRevision >= 786218 || browserRevision < 300000)) ||
-              (browserName === 'firefox' && browserRevision >= 1128) ||
-              (browserName === 'webkit' && browserRevision >= 1307) ||
-              // All new applications have a marker file right away.
-              (browserName !== 'firefox' && browserName !== 'chromium' && browserName !== 'webkit');
-          if (!shouldHaveMarkerFile || (await existsAsync(browserDirectoryToMarkerFilePath(usedBrowserPath))))
-            usedBrowserPaths.add(usedBrowserPath);
+
+          const browserPath = descriptor.dir;
+          const browserVersion = parseInt(descriptor.revision, 10);
+          browserList.push({
+            browserName,
+            browserVersion,
+            hostDir: linkTarget,
+            browserPath
+          });
         }
       } catch (e) {
         await fs.promises.unlink(linkPath).catch(e => {});
```

Comment:
`_traverse*` is a readonly method, it should not change filesystem. Maybe return it in a separate list of broken links? I'm not sure if we want to show it in `install --list` output, but at the very least `install --list` should not make changes to the filesystem.

Reply:
Totally aligned on ensuring no changes to filesystem. Thanks for flagging this out, I missed verifying the unlink logic. 
Will set a flag to denote listOnly boolean for unlink logic.